.github/workflows/_reusable.build-flutter-example.yml

```diff
@@ -13,15 +13,19 @@ permissions:
 
 jobs:
   flutter-build-example-ios:
-    runs-on: macos-11
+    runs-on: hetzner-mac-pm
     timeout-minutes: 30
     steps:
+      - name: Clean up workspace
+        shell: bash
+        working-directory: ${{ github.workspace }}
+        run: rm -rf *
```

Comment:
❔ also just to help me understand.. why we need this? it this because of that "leftover" issue between runs that you've mentioned during planning?

Reply:
yes 👍
If you use the github-hosted runner you always get a runner with a clean state. What I say now might not 100% accurate but Github runs its runner in a VM and restores the VM to a clean state after a job completes. Our self-hosted runner, on the other hand, run on the host without isolation. This means that we have to "manually" clean some parts of the runner between jobs. At the moment there is no easy way to clean up the jobs automatically via the runner. I think that's because Github runs the runner in the VM and therefore doesn't really need this feature. But Github now provides a way to run scripts before and after a job, so later in the future we can at least move that specific runner detail into such a script. The reason we only need to clean the workspace on the mac runner is because on linux we run the jobs in docker containers. So we just start with a clean container, run the steps and remove it after the job is done. There is one exception, however, that we have to empty the RUNNER_TOOL_CACH and RUNNER_TEMP folders, as these are mounted on the host system and are not automatically cleaned up after a job.